docs/core/node.md

```diff
@@ -1,13 +1,78 @@
+
+- [`main` function](#main-function)
+- [`start` command](#start-command)
 
 ## `main` function
 
-TODO
+The full-node client of any SDK application is built by running a `main` function. The client is generally named by appending the `-d` suffix to the application name (e.g. `appd` for an application named `app`), and the `main` function is defined in a `cmd/appd/main.go` file. Running this function creates an executable `.appd` that comes with a set of commands. For an app named `app`, the main command is [`appd start`](#start-command), which starts the full-node. 
+
+In general, developers will implement the `main.go` function with the following structure:
+
+- First, a [`codec`](./encoding.md) is instanciated for the application.
+- Then, the [`config`](https://github.com/cosmos/cosmos-sdk/blob/master/types/config.go) is retrieved and config parameters are set. This mainly involves setting the bech32 prefixes for [addresses and pubkeys](../basics/accounts-fees-gas.md#addresses-and-pubkeys).
+- Using [cobra](https://github.com/spf13/cobra), the root command of the full-node client is created. After that, all the custom commands of the application are added using the `AddCommand()` method of `rootCmd`. 
+- Add default server commands to `rootCmd` using the `server.AddCommands(ctx, cdc, rootCmd, newApp, exportAppStateAndTMValidators)` method. These commands are separated from the ones added above since they are standard and defined at SDK level. They should be shared by all SDK-based application. They include the most important command: the [`start` command](#start-command).
+- Prepare and execute the [`executor`](https://github.com/tendermint/tendermint/blob/master/libs/cli/setup.go#L75-L78).  
+
+See an [example of `main` function](https://github.com/cosmos/gaia/blob/master/cmd/gaiad/main.go) from the `gaia` application. 
 
 ## `start` command
 
-TODO
+The `start` command is defined in the `/server` folder of the Cosmos SDK. It is added to the root command of the full-node client in the [`main` function](#main-function) and called by the end-user to start their node:
+
+```go
+// For an example app named "app", the following command starts the full-node
+
+appd start
+```
+
+As a reminder, the full-node is composed of three conceptual layers: the networking layer, the consensus layer and the application layer. The first two are generally bundled together in an entity called the consensus engine, while the third is the state-machine defined with the help of the Cosmos SDK. Currently, the Cosmos SDK uses Tendermint as the default consensus engine, meaning the start command is implemented to boot up a Tendermint node. 
```

Comment:
```suggestion
As a reminder, the full-node is composed of three conceptual layers: the networking layer, the consensus layer and the application layer.  The first two are generally bundled together in `Tendermint Core`  while the third is the state-machine defined with the help of the Cosmos SDK.

Reply:
"an entity call the consensus engine" is vaue, might want to specify Tendermint Core